### ISSUE TITLE
ChaliceViewError Should Print Real Error

### ISSUE THREAD

**Issue by woodlandh** [awslabs/chalice#50]
**tl;dr** ChaliceViewError should print the underlying error message instead of hiding it (I think?)

[heavily edited]

By default, ChaliceViewError does not raise the underlying exception, it only prints "An internal server error occurred."
When app.debug = True, ChaliceViewError instead raises the underlying exception.

I think both of these features should be enabled at the same time. ChaliceViewError should print the real message (so that it is logged in CloudWatch) and then return a nicely-wrapped ChaliceViewError to the client. This way the developer can get errors logged without worrying about 'null' returns on bad requests.

I've had to do this to gain this functionality:
`
    except Exception as e:
        print(e)
        raise e
`

And still every single error just gets raised as

`
Traceback (most recent call last):
  File "/var/task/chalice/__init__.py", line 148, in __call__
    raise e
`
Which isn't helpful.

**Comment by woodlandh** [comment]
Found a work-around, maybe I was doing something wrong with get_item, I'm using table.query() and response['Items'][0] and I'm getting something back now. So that makes me think the original problem isn't Chalice.

Regardless, how does one configure Chalice to debug mode so that it doesn't wrap every error as ChaliceViewError?

**Comment by jamesls** [comment]
Your suggestion of printing/logging the exception on the case of error I think makes a lot of sense.
